WcaOnRails/app/models/competition.rb

```diff
@@ -1718,4 +1718,11 @@ def to_ics
     cal.publish
     cal
   end
+
+  def exempt_from_wca_dues?
+    world_or_continental_championship = (championships.map(&:championship_type) & Championship::MAJOR_CHAMPIONSHIP_TYPES).any?
+    multi_country_fmc_competition = events.length == 1 && events[0].fewest_moves? && Country::FICTIVE_IDS.include?(countryId)
```

Comment:
(Rails novice).

Accessing these fields here _feels_ weird to me. Perhaps these should be methods on the `Championship` and `Country` models instead?

Reply:
Could you please go into detail what feels weird here?

You're working in the context of the `Competition` model. There's absolutely nothing wrong with having a competition compute whether it is a multi localtion FMC comp.
It might be worth extracting that heuristic into its own separate function, but then it would still stay a part of `competition.rb`.

Reply:
I wasn't sure if `Championship::MAJOR_CHAMPIONSHIP_TYPES` and `Country::FICTIVE_IDS` are "private" fields per se.

After more thought, I agree that computing at this level (with the changes you suggested) makes sense.